### PR TITLE
Documentation reorganization

### PR DESCRIPTION
**Pull request by pet-mit**
This PR includes a reorganization of the documentation content structure, plus some extra content:
- The website now has a richer tree, most of the content was redispatched in the new pages:
  - the top-most categories are "user guide" (instead of "reference guide") and "developer guide" (instead of "build")
  - the "user guide" is now easier to read:
    - with a focus on Antares_Simulator's two main features: solver and ts-generator
    - the other features are briefly documented but marked as deprecated
    - each of the features now has a new "parameters" page
    - the solver now has a new "heuristics" page intended to document all the heuristics used by the solver
    - both features have "inputs, outputs and command line" pages
- The PDF generation action is now run on same branches as other actions, in order to check that the PDF builds correctly
- New README in the docs directory to explain how developers should maintain the docs 
- Some overall updates & harmonization
- There is still a lot of work to be done (at least the tasks marked with a "TODO" comment)

The resulting doc can be previewed in the readthedocs automatic check or here: https://antares-simulator--2024.org.readthedocs.build/en/2024/

### CODE REVIEW

**Review by a-zakir — APPROVED** [review]

File: docs/user-guide/03-getting_started.md
Comment (vargastat) [inline docs/user-guide/03-getting_started.md:1]: In the extensions we should probably mention AntaresEditObject (https://cran.r-project.org/web/packages/antaresEditObject/index.html) and AntaresReadObject (https://cran.r-project.org/web/packages/antaresRead/index.html) as their are much more used than AntaresViz but it is a good idea to mention AntaresViz too!

File: docs/user-guide/solver/02-inputs.md
Comment (vargastat) [inline docs/user-guide/solver/02-inputs.md:1]: Today the easiest way to update files locally is to use packageR intended for it AntaresRead and AntaresEdit...but Notepad is an option too :)

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+root of the project and running:
+```bash
+cd docs/pdf-doc-generation-with-sphinx
+bash create_pdf_doc.sh reference-guide.pdf
+```
+Sphinx will create a `reference-guide.pdf` file in `docs/pdf-doc-generation-with-sphinx`.
+
+### Doxygen
+[//]: # (TODO)
+_**This section is under construction**_
```
Comment (a-zakir) [inline docs/README.md]:
```suggestion
we strive to enrich code documentation as it evolves. The doxygen doc is generated automatically each time the develop branch is updated. It is hosted [here](https://antaressimulatorteam.github.io/Antares_Simulator/doxygen), and can also be generated locally with the follow command at the root of the project:

**doxygen-executable** docs/Doxyfile

If you'd like the same rendering as the one published checkout [doxygen-awesome-css](https://github.com/jothepro/doxygen-awesome-css) and do this step before:
```
git clone https://github.com/jothepro/doxygen-awesome-css.git
cd doxygen-awesome-css
git checkout v2.2.1
git apply ../docs/antares-simulator-doxygen.patch

```

File: docs/developer-guide/continuous-integration.md [hunk 2]
```diff
@@ -5,21 +5,21 @@ Here is a description of workflows with their associated status.
...
-| Ubuntu           | `ubuntu-system-deps-build.yml`  | Builds Antares Simulator's dependencies                                                 | [![Status][ubuntu_deps_build_svg]][ubuntu_deps_build_link] |
-| Centos7          | `centos7-system-deps-build.yml` | Builds Antares Simulator's dependencies                                                 | [![Status][centos_deps_build_svg]][centos_deps_build_link] |
+| OS      | .yml                            | Description                             | Status                                                     |
+|:--------|---------------------------------|-----------------------------------------|------------------------------------------------------------|
+| Ubuntu  | `ubuntu-system-deps-build.yml`  | Builds Antares Simulator's dependencies | [![Status][ubuntu_deps_build_svg]][ubuntu_deps_build_link] |
+| Centos7 | `centos7-system-deps-build.yml` | Builds Antares Simulator's dependencies | [![Status][centos_deps_build_svg]][centos_deps_build_link] |
```
Comment (a-zakir) [inline docs/developer-guide/continuous-integration.md]: **\*deps_build\*** workflows are deprecated. they don't succeed anymore

File: docs/user-guide/solver/02-inputs.md
Comment (vargastat) [inline docs/user-guide/solver/02-inputs.md:1]: It depends on for which version the doc is intended but in Changelog.md earlier it was mentioned that for 9.0.0 STS group can by dynamic. This information should appear here too for InputFiles /Storages

**Review by vargastat — COMMENTED** [review]
WoW @pet-mit  nice first shot, it seems  to me much clearer than current doc. I put some minors comments :)

File: docs/developer-guide/continuous-integration.md (hunk 2 above)
Comment (pet-mit) [reply docs/developer-guide/continuous-integration.md]: I'll remove them from docs. they should be removed from the workflows if they're useless

File: docs/user-guide/solver/02-inputs.md
Comment (pet-mit) [reply docs/user-guide/solver/02-inputs.md:1]: this is actually one page that I did not have time to update yet (I put a TODO on the top of the page), I only created it and put in it all information that already existed and that related to input files. I suggest we update this in a new PR

File: docs/README.md (hunk 1 above)
Comment (pet-mit) [reply docs/README.md]: thanks ! :)

File: docs/user-guide/solver/02-inputs.md
Comment (flomnes) [reply docs/user-guide/solver/02-inputs.md:1]: > It depends on for which version the doc is intended but in Changelog.md earlier it was mentioned that for 9.0.0 STS group can by dynamic. This information should appear here too for InputFiles /Storages

Dynamic groups will be introduced in v9.1.0

**Review by flomnes — APPROVED** [review]

File: docs/user-guide/solver/05-model.md [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+| $F_l^- \in \mathbb{R}^T_+$                     | power flow through $l$, from $d_l$ to $u_l$                                                             |
+| $F_l\in \mathbb{R}^T$                          | total power flow through $l$, $F_l=F_l^+-F_l^-$                                                         |
+| $\tilde{F}_t \in \mathbb{R}^T$                 | system flow snapshot at time $t$                                                                        |
+| $\gamma_l^+\in \mathbb{R}^T$                   | transmission cost through $l$, from $u_l$ to $d_l$. Proportional to the power flow                      |
+| $\gamma_l^-\in \mathbb{R}^T$                   | transmission cost through $l$, from $d_l$ to $u_l$. Proportional to the power flow                      |
+| $Z_l \in \mathbb{R}\_+$                        | overall impedance of $l$                                                                                |
+
+### Thermal units
+
+<<<<<<<< HEAD:docs/user-guide/solver/05-model.md
```
Comment (MartinBelthle) [inline docs/user-guide/solver/05-model.md]: It seems that this part of the page doesn't render correctly

**Review by MartinBelthle — COMMENTED** [review]
The folder `docs/optimisation` doesn't appear anymore. It contains a page on thermal heuristics that should be merged inside the new `heuristics` one and the other page could be removed.

File: docs/developer-guide/ortools-integration.md [hunk 4]
```diff
@@ -24,31 +25,36 @@ These are the solver names and enum, defined in the [OR-Tools API](https://githu
...
+| `cbc`                | CBC_MIXED_INTEGER_PROGRAMMING    | Free-to-use | [https://github.com/coin-or/Cbc](https://github.com/coin-or/Cbc)                                           |
+| `highs`              | HIGHS_MIXED_INTEGER_PROGRAMMING  | Free-to-use | [https://highs.dev](https://highs.dev)                                                                     |
+| `sirius` [^1]        | SIRIUS_MIXED_INTEGER_PROGRAMMING | Free-to-use | [https://github.com/rte-france/sirius-solver](https://github.com/rte-france/sirius-solver)                 |
+| `gurobi`             | GUROBI_MIXED_INTEGER_PROGRAMMING | Commercial  | [https://www.gurobi.com](https://www.gurobi.com)                                                           |
+| `cplex`              | CPLEX_MIXED_INTEGER_PROGRAMMING  | Commercial  | [https://www.ibm.com/fr-fr/analytics/cplex-optimizer](https://www.ibm.com/fr-fr/analytics/cplex-optimizer) |
+| `xpress`             | XPRESS_MIXED_INTEGER_PROGRAMMING | Commercial  | [https://www.fico.com/en/products/fico-xpress-solver](https://www.fico.com/en/products/fico-xpress-solver) |
```
Comment (flomnes) [inline docs/developer-guide/ortools-integration.md]: I think only Sirius, XPRESS, SCIP and CBC are actually accessible from Antares Simulator

File: docs/developer-guide/ortools-integration.md [hunk 5]
```diff
@@ -24,31 +25,36 @@ These are the solver names and enum, defined in the [OR-Tools API](https://githu
...
-|:---------------------|-------------------------------------|-------------|-------------------|
-| `bop`                | BOP_INTEGER_PROGRAMMING             | Free-to-use | Boolean           |
-| `sat`                | SAT_INTEGER_PROGRAMMING             | Free-to-use | Boolean integer   |
-| `knapsack`           | KNAPSACK_MIXED_INTEGER_PROGRAMMING  | Free-to-use |                   |
+
+| OR-Tools solver name | Enum                               | Usage       | Optimization type |
+|:---------------------|------------------------------------|-------------|-------------------|
+| `bop`                | BOP_INTEGER_PROGRAMMING            | Free-to-use | Boolean           |
+| `sat`                | SAT_INTEGER_PROGRAMMING            | Free-to-use | Boolean integer   |
+| `knapsack`           | KNAPSACK_MIXED_INTEGER_PROGRAMMING | Free-to-use |                   |
```
Comment (flomnes) [inline docs/developer-guide/ortools-integration.md:45]: Constraint optimization solver can't be used from Antares Simulator as of now

**Comment by flomnes** [comment]
> The folder `docs/optimisation` doesn't appear anymore. It contains a page on thermal heuristics that should be merged inside the new `heuristics` one and the other page could be removed.

Fixed

**Pull request merged**